### PR TITLE
fix: unexpected lg name replacement in dialog dataTime

### LINKED ISSUE #2765: Date gets altered in body of Post Request
## Describe the bug
When adding a date using dateFormatTime() in the body of a Post Request, the - are getting replaced by _

## Version
Cannot see SHA in about page

## Browser

- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [X ] Edge

## OS

- [ ] macOS
- [X ] Windows
- [ ] Ubuntu

## To Reproduce

1. Add a Send a HTTP request component in the flow
    Method -> POST
    Url -> https://graph.microsoft.com/v1.0/me/events
    Body -> 
{
  "date": "${formatDateTime(utcNow(), 'yyyy-MM-ddTHH:mm')}",
  "date2": "2020-04-24T14:22"
}
2. Run the flow
3. Check the HttpRequest trace: value -> request -> content
4. Notice that the first date has _ in it, the second one is unaffected

## Expected behavior

Date with 202-04-02 format instead of 2020_04_02

### PR DESCRIPTION
**Pull request by zhixzhan**
## Task Item
close #2765

### CODE REVIEW

File: Composer/packages/server/src/models/bot/botProject.ts [hunk 1]
```diff
@@ -644,7 +644,6 @@ export class BotProject {
                 replacers.map(replacer => {
                   newLine = replacer(newLine);
                 });
-                newLine = newLine.replace('-', '_');
                 newContentLines.push(newLine);
```
Comment (boydc2014) [inline Composer/packages/server/src/models/bot/botProject.ts:647]: so the fix is just don't replace?

Comment (zhixzhan) [reply Composer/packages/server/src/models/bot/botProject.ts:647]: Fixed by only replace items defined in replacers(above line), this line is too generic and risky. By remove this line, the replacement in *.dialog *.lg would be same. 
includes 
```
bfdactivity-1234 -> SendActivity_1234
bfdprompt-1234 -> TextInput_Prompt_1234
bfdinvalidPrompt-1234 -> TextInput_InvalidPrompt_1234
bfdunrecognizedPrompt-1234 -> TextInput_UnrecognizedPrompt_1234
bfddefaultValueResponse-1234 -> TextInput_DefaultValueResponse_1234
```
excludes described in #2597 

Won't auto-replace case:

reference is user customize defined lg template name,
- ${my-template()}
technically can not distinguish it is an expression or a template reference.

"

**Comment by a-b-r-o-w-n** [comment]
I'd like to see some unit tests for this. I'll go ahead and merge, @zhixzhan can you add unit tests in a follow-up PR?

**Comment by zhixzhan** [comment]
> I'd like to see some unit tests for this. I'll go ahead and merge, @zhixzhan can you add unit tests in a follow-up PR?

Okay

**Pull request merged**